scrypto/src/values.rs

```diff
@@ -413,13 +461,30 @@ impl CustomValueVisitor for ScryptoCustomValueChecker {
 /// Utility that formats any Scrypto value.
 pub struct ScryptoValueFormatter {}
 
+pub struct ScryptoValueFormatterContext<'a> {
+    bech32_encoder: Option<&'a Bech32Encoder>,
+    bucket_ids: &'a HashMap<BucketId, String>,
+    proof_ids: &'a HashMap<ProofId, String>,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum ScryptoValueFormatterError {
+    InvalidTypeId(u8),
+    InvalidCustomData(ScryptoCustomValueCheckError),
+}
```

Comment:
The original intention of `ScryptoValueFormatter` is to format `ScryptoValue` (validated) so we should panic if unable to format, so to help identify implementation bug. 

With many of the methods in the formatter being public and can be used for other use cases, I'm also okay with this error code approach.